### ISSUE TITLE
Office 365 not connecting!!!

### ISSUE THREAD

**Issue by rhadw** [getmail6/getmail6#144]
Hi.
I have a working config which worked for 2 years.
Since the beginning of the month, getmail can't connect to my O365 account to fetch the emails but I don't seem to make it work.
What I have done untill now:
1. Created an app pass for that account but no luck, it doesn't connect and in Azure sign in logs I can see the failures: "User did not pass the MFA challenge (non interactive)."
In getmail's log, the only thing is 'getmailrc: credential/login error (b'LOGIN failed.')'
3. I've tried creating an app registration but that's overly complicated and I didn't went through with all of the steps so I couldn't use this option.
I believe all these problems have to do with this article from Microsoft, from the beginning of the month.
https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online
Basically, as far as I understand, getmail does basic authentication but Microsoft wants some MFA/tokens authentication.

Any ideas, please?
Radu

**Comment by rpuntaie** [comment]
#145 has some overlap with this. Basically someone who has a Microsoft account needs to find a solution and create a pull request.

**Comment by ykasap** [comment]
@rhadw You cannot use an app password because it still uses Basic Authentication (PLAIN/LOGIN), which was retired in Oct. 2022. You must complete all the steps registering getmail6 in your Azure AD to use OAuth 2.0 with getmail6.

**Comment by rpuntaie** [comment]
With @ykasp explanation and #146, I close this, too.
